src/content/docs/en/extend/add-view-transitions.mdx

```diff
@@ -0,0 +1,353 @@
+            return "dark";
+          }
+          return "light";
+        })();
+
+        if (theme === "light") {
+          document.documentElement.classList.remove("dark");
+        } else {
+          document.documentElement.classList.add("dark");
+        }
+
+        window.localStorage.setItem("theme", theme);
+
+        const handleToggleClick = () => {
+          const element = document.documentElement;
+          element.classList.toggle("dark");
+
+          const isDark = element.classList.contains("dark");
+          localStorage.setItem("theme", isDark ? "dark" : "light");
+        };
+
+        document
+          .getElementById("themeToggle")
+          .addEventListener("click", handleToggleClick);
+      });
+    </script>
+    ```
+    Now, the theme toggle is interactive on every page when using the `<ViewTransitions />` router, after the page has finished loading.
+
+6. Check for theme earlier to prevent flashing in dark mode.
+
+    The theme toggle works on every page, but its script is loaded at the end of the navigation process, **after the new page has fully loaded in the browser**. There may be a flash of the light theme version of the site before this theme toggle script runs and checks which theme it should use on the page.
+
+    To check for, and if necessary set, dark mode earlier in the navigation process, create a function that will run in response to the `astro:after-swap` event. The following function to check the browser's `localStorage` for dark theme will run **immediately after the new page has replaced the old page**, before the DOM elements are painted to the screen.
+
+    Add this new script to the `<ThemeIcon />` component, in addition to the script that controls the theme toggle.
+
+    ```astro title="src/components/ThemeIcon.astro" ins={3-9}
+    <script> ... <script>
+
+    <script>
+      document.addEventListener('astro:after-swap', () => {
+        localStorage.theme === 'dark' 
+        ? document.documentElement.classList.add("dark")
+        : document.documentElement.classList.add("light");
+      }); 
+    </script>
+    ```
+
+    Now, every page change **that uses the `<ViewTransitions />` router for client-side navigation** (and therefore access to the `astro:after-swap` event) will be able to detect `theme: dark` from the browser's `localStorage` and update the current page accordingly before the page is rendered for the viewer.
+
+### Customize Transition animations
+
+7. Change the default `fade` animation to a `slide` for the page title.
+
+    With view transitions enabled, you currently have a small fade in and out set for all page transition animations. Astro also provides a built-in `slide` animation. To change the type of animation for a single element, add the `transition:animation=""` directive.
+
+    For example, to make the page titles slide in instead of fade in, add `transition:animate="slide"` to the `<h1>` element in the BaseLayout:
```

Comment:
There are several places here we are using "fade in", maybe with the exception of the first use (paragraph above), I believe all these should be "fade-in" (with a hyphen) instead.

Reply:
I think I'm using it as a verb everywhere, not a noun, so I feel pretty confident these are correct as is. (Admittedly there may be one or two places where either verb or noun could be correct, but I checked them all and there isn't anywhere I'd clearly think "noun")